### PR TITLE
[KYUUBI-280] Align Operation GET_SCHEMAS * behavior with Hive

### LINKED ISSUE #280: CDH 6.3.1 HUE can not list database
**Describe the bug**
CDH 6.3.1 HUE can not list database

**To Reproduce**
1. Configurations
2. Environments
  - CDH 6.3.1 HUE
  - Kyuubi master code
  - Spark 3.0.1
3. Operations

**Expected behavior**

**Screenshots**

image

**Additional context**

```
2021-01-13 20:29:29.276 DEBUG service.FrontendService: TGetSchemasReq(sessionHandle:TSessionHandle(sessionId:THandleIdentifier(guid:B9 D7 B7 A2 56 3F 4A BE 9D 6A EA 69 4F 79 F6 F2, secret:19 A6 9A 9C 9B 61 4D 74 8B C1 46 70 3C 86 95 C8)), schemaName:*)
2021-01-13 20:29:29.276 INFO operation.GetSchemas: Processing bigdata's query[6ec6f36a-5d76-43d7-8e39-47a1f9930c18]: INITIALIZED_STATE -> RUNNING_STATE, statement: GET_SCHEMAS
2021-01-13 20:29:29.282 INFO operation.GetSchemas: Processing bigdata's query[6ec6f36a-5d76-43d7-8e39-47a1f9930c18]: RUNNING_STATE -> ERROR_STATE, statement: GET_SCHEMAS, time taken: 0.006 seconds
2021-01-13 20:29:29.283 WARN service.FrontendService: Error getting schemas:
org.apache.kyuubi.KyuubiSQLException: Error operating GET_SCHEMAS: java.util.regex.PatternSyntaxException: Dangling meta character '*' near index 0
*
^
	at java.util.regex.Pattern.error(Pattern.java:1957)
	at java.util.regex.Pattern.sequence(Pattern.java:2125)
	at java.util.regex.Pattern.expr(Pattern.java:1998)
	at java.util.regex.Pattern.compile(Pattern.java:1698)
	at java.util.regex.Pattern.<init>(Pattern.java:1351)
	at java.util.regex.Pattern.compile(Pattern.java:1028)
	at org.apache.kyuubi.engine.spark.operation.GetSchemas.runInternal(GetSchemas.scala:48)
	at org.apache.kyuubi.operation.AbstractOperation.run(AbstractOperation.scala:127)
	at org.apache.kyuubi.session.AbstractSession.runOperation(AbstractSession.scala:91)
	at org.apache.kyuubi.session.AbstractSession.getSchemas(AbstractSession.scala:156)
	at org.apache.kyuubi.service.AbstractBackendService.getSchemas(AbstractBackendService.scala:84)
	at org.apache.kyuubi.service.FrontendService.GetSchemas(FrontendService.scala:291)
	at org.apache.hive.service.rpc.thrift.TCLIService$Processor$GetSchemas.getResult(TCLIService.java:1497)
	at org.apache.hive.service.rpc.thrift.TCLIService$Processor$GetSchemas.getResult(TCLIService.java:1482)
	at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:38)
	at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:39)
	at org.apache.kyuubi.service.authentication.TSetIpAddressProcessor.process(TSetIpAddressProcessor.scala:36)
	at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:310)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

	at org.apache.kyuubi.KyuubiSQLException$.apply(KyuubiSQLException.scala:51)
	at org.apache.kyuubi.engine.spark.operation.SparkOperation$$anonfun$onError$1.applyOrElse(SparkOperation.scala:96)
	at org.apache.kyuubi.engine.spark.operation.SparkOperation$$anonfun$onError$1.applyOrElse(SparkOperation.scala:87)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:38)
	at org.apache.kyuubi.engine.spark.operation.GetSchemas.runInternal(GetSchemas.scala:54)
	at org.apache.kyuubi.operation.AbstractOperation.run(AbstractOperation.scala:127)
	at org.apache.kyuubi.session.AbstractSession.runOperation(AbstractSession.scala:91)
	at org.apache.kyuubi.session.AbstractSession.getSchemas(AbstractSession.scala:156)
	at org.apache.kyuubi.service.AbstractBackendService.getSchemas(AbstractBackendService.scala:84)
	at org.apache.kyuubi.service.FrontendService.GetSchemas(FrontendService.scala:291)
	at org.apache.hive.service.rpc.thrift.TCLIService$Processor$GetSchemas.getResult(TCLIService.java:1497)
	at org.apache.hive.service.rpc.thrift.TCLIService$Processor$GetSchemas.getResult(TCLIService.java:1482)
	at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:38)
	at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:39)
	at org.apache.kyuubi.service.authentication.TSetIpAddressProcessor.process(TSetIpAddressProcessor.scala:36)
	at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:310)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.RuntimeException: java.util.regex.PatternSyntaxException:Dangling meta character '*' near index 0
*
^
	at java.util.regex.Pattern.error(Pattern.java:1957)
	at java.util.regex.Pattern.sequence(Pattern.java:2125)
	at java.util.regex.Pattern.expr(Pattern.java:1998)
	at java.util.regex.Pattern.compile(Pattern.java:1698)
	at java.util.regex.Pattern.<init>(Pattern.java:1351)
	at java.util.regex.Pattern.compile(Pattern.java:1028)
	at org.apache.kyuubi.engine.spark.operation.GetSchemas.runInternal(GetSchemas.scala:48)
```

### PR DESCRIPTION
**Pull request by pan3793**
pan3793 [PR 292](https://github.com/yaooqinn/kyuubi/pull/292) Bug [&#10088;?&#10089;](https://pullrequestbadge.com/?utm_medium=github&utm_source=yaooqinn&utm_campaign=badge_info)

### Please add issue ID here?

Fixes #280

### Why are the changes needed?

Before this PR, Kyuubi treat Operation GET_SCHEMAS * as an invalid Operation then cause HUE list databases failed, but HiveServer2 will return all databases on same request.

### Test Plan:
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible
- [x] Add screenshots for manual tests if appropriate
- [x] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

After PR:
201610862635_ pic

### CODE REVIEW

**Comment by yaooqinn** [comment]
Is it the only place we encounter with this issue?  If so, I can accept such a hard code change.

**Comment by pan3793** [comment]
@yaooqinn In my use case, so far, I have only encountered this one issue of integration with CDH 6.3.1 HUE.

**Comment by yaooqinn** [comment]
if it is related to the schema pattern, we can not do such thing like this.  this breaks the consistency among meta data operations.  if we do here, we need do for others.

**Comment by pan3793** [comment]
I don't think it's related to the schema pattern. The original pattern match code almost copy from Hive(I guess), and 
1. In `org.apache.hadoop.hive.metastore.ObjectStore#getDatabase`, the `pattern` is the converted pattern
2. I did not see other methods like `getTables`, `getFunctions` has the same exception for `*`

**Comment by pan3793** [comment]
Wait... I think I need some more check on `org.apache.kyuubi.operation.JDBCTests#183`
```
val e = intercept[HiveSQLException](metaData.getTables(null, "*", null, null))
assert(e.getCause.getMessage contains "Dangling meta character '*' near index 0\n*\n^")
```

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetSchemas.scala [hunk 1]
```diff
@@ -44,8 +45,12 @@ class GetSchemas(spark: SparkSession, session: Session, catalogName: String, sch
       val schemaPattern = convertSchemaPattern(schema)
       val databases = spark.sessionState.catalog.listDatabases(schemaPattern)
       val globalTmpViewDb = spark.sessionState.catalog.globalTempViewManager.database
-      if (Pattern.compile(convertSchemaPattern(schema, false))
-          .matcher(globalTmpViewDb).matches()) {
+      if (
+        // Hive Metastore treat empty schema or * as get all databases,
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetSchemas.scala:49]: How about not make comment like this?

Comment (pan3793) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetSchemas.scala]: removed

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetColumns.scala [hunk 2]
```diff
@@ -17,10 +17,12 @@
 
 package org.apache.kyuubi.engine.spark.operation
 
+
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetColumns.scala:20]: nit: unnecessary

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala [hunk 3]
```diff
@@ -137,9 +137,6 @@ class SparkOperationSuite extends WithSparkSQLEngine {
         assert(pos === 18, "all columns should have been verified")
       }
 
-      val e = intercept[HiveSQLException](metaData.getColumns(null, "*", null, null))
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala:140]: shall we add a positive test here instead?

**Review by yaooqinn — APPROVED** [review]

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetColumns.scala (hunk 2 above)
Comment (pan3793) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/GetColumns.scala]: removed

File: externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala (hunk 3 above)
Comment (pan3793) [reply externals/kyuubi-spark-sql-engine/src/test/scala/org/apache/kyuubi/engine/spark/operation/SparkOperationSuite.scala:140]: done

**Comment by yaooqinn** [comment]
thanks. merged to master for 1.1

**Comment by yaooqinn** [comment]
Hi @pan3793, would you mind creating a quick start guide for HUE users?
Here is an example https://kyuubi.readthedocs.io/en/latest/quick_start/quick_start_with_dbeaver.html

**Comment by pan3793** [comment]
> would you mind creating a quick start guide for HUE users?

I was trying to do it, but encountered HUE docker image issue on last Friday. Maybe will done it in this week.

**Comment by yaooqinn** [comment]
Wonderful!
